src/Wallet/Cip30.purs

```diff
@@ -82,14 +81,15 @@ getWalletAddress :: Cip30Connection -> Aff (Maybe Address)
 getWalletAddress conn = fromHexString _getAddress conn <#>
   (_ >>= addressFromBytes <<< rawBytesAsCborBytes)
 
-getCollateral :: Cip30Connection -> Aff (Maybe TransactionUnspentOutput)
-getCollateral conn = fromMaybeHexString getCip30Collateral conn >>=
+getCollateral :: Cip30Connection -> Aff (Maybe (Array TransactionUnspentOutput))
+getCollateral conn = fromMaybeHexStrings getCip30Collateral conn >>=
   case _ of
     Nothing -> pure Nothing
-    Just bytes -> do
-      liftEffect $
-        Deserialization.UnspentOuput.convertUnspentOutput
-          <$> fromBytesEffect (unwrap bytes)
+    Just collateralUtxos -> liftEffect do
+      Just <$> for collateralUtxos \bytes -> do
+        maybe (throw "Unable to convert UTxO") pure =<<
+          Deserialization.UnspentOuput.convertUnspentOutput
+            <$> fromBytesEffect (unwrap bytes)
```

Comment:
```suggestion
    \bytes -> do
      utxo <- liftEffect $ fromBytesEffect $ unwrap bytes
      let utxo' = Deserialization.UnspentOuput.convertUnspentOutput utxo
      liftMaybe (error "Unable to convert UTxO") utxo'
```

Reply:
I don't personally find this more readable

Reply:
Fair enough

Reply:
At minimum I think `fromMaybeHexString` is doing more than it needs to, `<<< act` could be completely removed and the application of Cip30Connection moved to the call site

Reply:
I refactored it, but another way.